src/components/Review/AddReview.js

```diff
@@ -0,0 +1,147 @@
+import React from "react";
+import { withStyles } from "@material-ui/core/styles";
+import TextField from "@material-ui/core/TextField";
+import Grid from "@material-ui/core/Grid";
+import Paper from "@material-ui/core/Paper";
+import Typography from "@material-ui/core/Typography";
+import Rating from "@material-ui/lab/Rating";
+import Button from "@material-ui/core/Button";
+import { connect } from "react-redux";
+import { addReview } from "../../redux/actions/userDetailActions";
+import { closeModal } from "../../redux/actions/modalActions";
+
+const useStyles = (theme) => ({
+  paper: {
+    width: "100%",
+    backgroundColor: theme.palette.background.paper,
+    boxShadow: theme.shadows[5],
+    padding: theme.spacing(2, 4, 3),
+    outline: "none",
+    position: "relative",
+  },
+  ratingContainer: {
+    display: "flex",
+    alignItems: "center",
+    justifyContent: "flex-start",
+    marginRight: 0,
+  },
+  reviewContainer: {
+    display: "inline-block",
+    width: "100%",
+    minHeight: "270px",
+    padding: theme.spacing(4, 0),
+  },
+  buttonHeader: {
+    display: "flex",
+    justifyContent: "space-between",
+    margin: theme.spacing(1),
+  },
+  cancelButton: {
+    color: "#AD343E",
+    borderColor: "#AD343E",
+  },
+  rating: {
+    marginLeft: theme.spacing(2),
+  },
+});
+
+class ReviewModal extends React.Component {
+  constructor(props) {
+    super(props);
+    this.state = {
+      title: "",
+      rating: 0,
+      review: "",
+    };
+  }
+
+  handleInputChange = (e) => {
+    let { name } = e.target;
+    let value = name === "rating" ? parseInt(e.target.value) : e.target.value;
+    this.setState({ [name]: value });
+  };
+
+  handleSubmit = () => {
+    let review = {
+      ...this.state,
+      reviewerUsername: this.props.currentUser.userName,
```

Comment:
Not really related to this commit, but this reminds me: @abid-salahi  did we have a check to ensure that usernames are unique when users are being created? :O